src/main/steps/existing-application/post.test.ts

```diff
@@ -67,32 +74,71 @@ describe('ExistingApplicationPostController', () => {
   test.each([
     [ApplicationType.JOINT_APPLICATION, '2022-08-22', undefined],
     [ApplicationType.SOLE_APPLICATION, '2022-08-22', '2022-09-15'],
-  ])('Should be redirected to the current page for %s', async (applicationType, dateSubmitted, dateAosSubmitted) => {
+  ])(
+    'When New case chosen but unlinking not allowed (%s)',
+    async (applicationType, dateSubmitted, dateAosSubmitted) => {
+      const body = {
+        existingOrNewApplication: existingOrNew.New,
+      };
+
+      const caseApiMockFnReturnValue = caseApiMockFn({ applicationType, dateSubmitted, dateAosSubmitted, id: '1234' });
+      (getCaseApiMock as jest.Mock).mockReturnValue(caseApiMockFnReturnValue);
+      const req = mockRequest({ body });
+      req.url = EXISTING_APPLICATION;
+      const res = mockResponse();
+
+      const controller = new ExistingApplicationPostController(mockFormContent.fields);
+      await controller.post(req, res);
+
+      expect(res.redirect).toHaveBeenCalledWith(EXISTING_APPLICATION);
+      expect(req.session.cannotLinkToNewCase).toBeTruthy();
+      expect(req.session.existingApplicationType).toBe(applicationType);
+      expect(caseApiMockFnReturnValue.triggerEvent).toHaveBeenCalledWith(
+        req.session.inviteCaseId,
+        {},
+        SYSTEM_CANCEL_CASE_INVITE
+      );
+    }
+  );
+
+  test('When New case chosen and user is respondent on submitted but not AOS submitted existing case', async () => {
```

Comment:
do we have a test when respondent is allowed to unlink?

Reply:
done